qubes/events.py

```diff
@@ -157,50 +160,92 @@ def _fire_event_in_order(self, order, event, kwargs):
                     key=(lambda handler: hasattr(handler, 'ha_bound')),
                     reverse=True):
                 effect = func(self, event, **kwargs)
+                # "is True" to rule out unittest.mock.Mock...
+                if getattr(func, 'ha_async', False) is True:
+                    assert effect is not None
```

Comment:
I'd very much prefer `assert inspect.isawaitable(effect)` instead of `is not None`. `*.iscoroutine()` may not be enough, since someone might like to return a plain `Future` (not a `Task`) or whatever.